private/bufpkg/bufconfig/write.go

```diff
@@ -273,30 +257,173 @@ func writeConfig(
+		for i, dependencyModuleReference := range writeConfigOptions.dependencyModuleReferences {
+			dependencies[i] = dependencyModuleReference.String()
+		}
 	}
-	tmplData := tmplUndocumentedData
 	if writeConfigOptions.documentationComments {
-		tmplData = tmplDocumentationCommentsData
+		// Write out config using template with document comments.
+		return writeCommentedConfig(ctx, writeBucket, config, dependencies, writeConfigOptions.uncomment)
+	}
+	// Write out raw default config without comments using externalConfig{V1, V1Beta1}
+	return writeExternalConfig(ctx, writeBucket, version, config, dependencies)
+}
+
+func writeExternalConfig(
+	ctx context.Context,
+	writeBucket storage.WriteBucket,
+	version string,
+	config *Config,
+	dependencies []string,
+) error {
+	var name string
+	if config.ModuleIdentity != nil {
+		name = config.ModuleIdentity.IdentityString()
+	}
+	breakingConfig := config.Breaking
+	lintConfig := config.Lint
+	switch version {
+	case V1Beta1Version:
+		var externalBreakingConfig bufbreakingconfig.ExternalConfigV1Beta1
+		if breakingConfig != nil {
+			externalBreakingConfig = bufbreakingconfig.ExternalConfigV1Beta1ForConfig(breakingConfig)
+		}
+		var externalLintConfig buflintconfig.ExternalConfigV1Beta1
+		if lintConfig != nil {
+			externalLintConfig = buflintconfig.ExternalConfigV1Beta1ForConfig(lintConfig)
+		}
+		externalConfig := ExternalConfigV1Beta1{
+			Name:     name,
+			Version:  config.Version,
+			Deps:     dependencies,
+			Breaking: externalBreakingConfig,
+			Lint:     externalLintConfig,
+		}
+		var buffer bytes.Buffer
+		encoder := yaml.NewEncoder(&buffer)
+		encoder.SetIndent(2)
+		if err := encoder.Encode(externalConfig); err != nil {
+			return err
+		}
+		return storage.PutPath(ctx, writeBucket, ExternalConfigV1Beta1FilePath, buffer.Bytes())
+	case V1Version:
+		var externalBreakingConfig bufbreakingconfig.ExternalConfigV1
+		if breakingConfig != nil {
+			externalBreakingConfig = bufbreakingconfig.ExternalConfigV1ForConfig(breakingConfig)
+		}
+		var externalLintConfig buflintconfig.ExternalConfigV1
+		if lintConfig != nil {
+			externalLintConfig = buflintconfig.ExternalConfigV1ForConfig(lintConfig)
+		}
+		externalConfig := ExternalConfigV1{
+			Name:     name,
+			Version:  config.Version,
+			Deps:     dependencies,
+			Breaking: externalBreakingConfig,
+			Lint:     externalLintConfig,
+		}
+		var buffer bytes.Buffer
+		encoder := yaml.NewEncoder(&buffer)
+		encoder.SetIndent(2)
+		if err := encoder.Encode(externalConfig); err != nil {
+			return err
+		}
+		return storage.PutPath(ctx, writeBucket, ExternalConfigV1FilePath, buffer.Bytes())
+	default:
+		return fmt.Errorf(`%s has an invalid "version %s"`, name, version)
+	}
+}
+
+func validateVersion(version string) error {
+	switch version {
+	case V1Version, V1Beta1Version:
+		return nil
+	default:
+		return fmt.Errorf("invalid config version %q provided", version)
+	}
+}
```

Comment:
nit: Move this function down so that it's adjacent to the other validation helper (at the bottom of the file),